src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.cpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp"
+
+#include <algorithm>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/DgSubcell/PerssonTci.hpp"
+#include "Evolution/Systems/NewtonianEuler/PrimitiveFromConservative.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace NewtonianEuler::subcell {
+template <size_t Dim>
+template <size_t ThermodynamicDim>
+bool TciOnFdGrid<Dim>::apply(
+    const gsl::not_null<Variables<
+        tmpl::list<MassDensity, Velocity, SpecificInternalEnergy, Pressure>>*>
+        subcell_grid_prim_vars,
+    const Scalar<DataVector>& subcell_mass_density,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& subcell_momentum_density,
+    const Scalar<DataVector>& subcell_energy_density,
+    const Scalar<DataVector>& dg_mass_density,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& dg_momentum_density,
+    const Scalar<DataVector>& dg_energy_density,
+    const EquationsOfState::EquationOfState<false, ThermodynamicDim>& eos,
+    const Mesh<Dim>& dg_mesh, const double persson_exponent) noexcept {
+  NewtonianEuler::PrimitiveFromConservative<Dim, ThermodynamicDim>::apply(
+      make_not_null(&get<MassDensity>(*subcell_grid_prim_vars)),
+      make_not_null(&get<Velocity>(*subcell_grid_prim_vars)),
+      make_not_null(&get<SpecificInternalEnergy>(*subcell_grid_prim_vars)),
+      make_not_null(&get<Pressure>(*subcell_grid_prim_vars)),
+      subcell_mass_density, subcell_momentum_density, subcell_energy_density,
+      eos);
+  Variables<tmpl::list<MassDensity, Velocity, SpecificInternalEnergy, Pressure>>
+      dg_grid_prim_vars{get(dg_energy_density).size()};
+  NewtonianEuler::PrimitiveFromConservative<Dim, ThermodynamicDim>::apply(
+      make_not_null(&get<MassDensity>(dg_grid_prim_vars)),
+      make_not_null(&get<Velocity>(dg_grid_prim_vars)),
+      make_not_null(&get<SpecificInternalEnergy>(dg_grid_prim_vars)),
+      make_not_null(&get<Pressure>(dg_grid_prim_vars)), dg_mass_density,
+      dg_momentum_density, dg_energy_density, eos);
+
+  using std::min;
+  return min(min(get(subcell_mass_density)), min(get(dg_mass_density))) <
+             min_density_allowed or
+         min(min(get(get<Pressure>(*subcell_grid_prim_vars))),
+             min(get(get<Pressure>(dg_grid_prim_vars)))) <
+             min_pressure_allowed or
+         evolution::dg::subcell::persson_tci(dg_mass_density, dg_mesh,
+                                             persson_exponent, 1.0e-18) or
```

Comment:
Can you make a `constexpr double persson_tci_epsilon = 1.0e-18;` as in the other TCI?